src/tunnel_controller.rs

```diff
@@ -89,3 +96,52 @@ impl Tunnel {
         }
     }
 }
+
+
+#[cfg(test)]
+describe! tunnel {
+
+    before_each {
+        use std::thread::sleep;
+        use std::time::Duration;
+        use stubs::tunnel::TunnelConfigStub;
+
+        let config = TunnelConfigStub::new(1, "a".to_owned());
+        let mut tunnel = super::Tunnel::new(config.clone());
+
+        tunnel.start().unwrap();
+        // XXX Sleep needed otherwise the process is not started in time
```

Comment:
I'm not too sure why a wait is needed, when start waits on the process to be spawned and [so does ManagedProcess](https://github.com/fxbox/foxbox/blob/e78390d2b46447805a1c316a110075af27826970/src/managed_process.rs#L93). Do you have an idea of what I could change in the Stub, @samgiles ?